src/BizHawk.Client.EmuHawk/RetroAchievements/LibRCheevos.cs

```diff
@@ -104,8 +104,8 @@ public struct rc_api_request_t
 			public IntPtr post_data;
 			public rc_api_buffer_t buffer;
 
-			public string URL => Mershul.PtrToStringUtf8(url);
-			public string PostData => Mershul.PtrToStringUtf8(post_data);
+			public readonly string URL => Mershul.PtrToStringUtf8(url);

```

Comment:
These are properties (=> means getter only), so this should just be completely invalid and error on build? How is this even valid (and what would it even mean here?)

Reply:
I believe that the => getter only syntax does not actually enforce that the getter does not mutate state, it is theoretically possible though terrible practice to mutate state inside a getter. The readonly keyword actually enforces that the property is not modified at the compiler level.

Reply:
How would the property ever get "modified" anyway? Isn't `=>` basically shorthand for `{ get { return ...} }`?

Reply:
If I were to modify a property in a class, I would get "Property cannot be 'readonly'".
![image](https://github.com/TASEmulators/BizHawk/assets/50538166/fcd7e5fc-a241-462d-b371-3e660af3ceba)
However, if it's in a struct, somehow it is allowed?
![image](https://github.com/TASEmulators/BizHawk/assets/50538166/eb7b0b78-2036-481f-851e-09b916d2c2e4)

Reply:
Seems like it may be due to this: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#readonly-instance-members?

Reply:
So I guess making struct properties readonly is like a guarantee to the caller that the struct itself won't get modified by the call... probably fine, although it shouldn't be required for our code.
Maybe making the entire struct readonly instead would be even better in this case?

Reply:
The structs in this case shouldn't be readonly as they get passed by ref in interop.

Reply:
Yes, this is a simple way of marking a function (prop getter) as pure. Why it's not allowed in classes I have no idea.